indico/modules/events/registration/models/registrations.py

```diff
@@ -825,3 +829,13 @@ def _set_checked_in_dt(target, value, *unused):
     @listens_for(Registration.transaction, 'set')
     def _set_transaction_id(target, value, *unused):
         value.registration = target
+
+    query = (select([db.func.sum(db.func.jsonb_array_length(RegistrationData.data)) + 1])
+             .where(db.and_((RegistrationData.registration_id == Registration.id),
+                            (RegistrationData.field_data_id == RegistrationFormFieldData.id),
+                            (RegistrationFormFieldData.field_id == RegistrationFormItem.id),
+                            (RegistrationFormItem.input_type == 'accompanying_persons'),
+                            db.cast(RegistrationFormItem.data['persons_count_against_limit'], db.Boolean)))
+             .correlate_except(RegistrationData)
+             .scalar_subquery())
+    Registration.occupied_slots = column_property(query, deferred=True)
```

Comment:
Just to make sure the query is correct, use the db log util to have a look at the query that gets executed when:

- accessing the property on a Registration instance
- querying a registration (or multiple) with `.options(undefer('occupied_slots'))`

A unit test is probably a good idea as well :)